crates/uniffi/uniffi_SPLIT_ME/src/system/sargon_os/add_factor_source/device_mnemonic_builder.rs

```diff
@@ -0,0 +1,247 @@
+use crate::prelude::*;
+use sargon::DeviceMnemonicBuildResult as InternalDeviceMnemonicBuildResult;
+use sargon::DeviceMnemonicBuilder as InternalDeviceMnemonicBuilder;
+
+/// A builder of `MnemonicWithPassphrase` required for a new `DeviceFactorSource` creation.
+/// Exposes functions to be called by hosts to be able to use the resulting `MnemonicWithPassphrase`.
+#[derive(Debug, Hash, PartialEq, Clone, uniffi::Object)]
+#[uniffi::export(Hash, Eq)]
+pub struct DeviceMnemonicBuilder {
+    wrapped: Arc<InternalDeviceMnemonicBuilder>,
+}
+
+/// The result of the `build` function from `DeviceMnemonicBuilder`.
+#[derive(Debug, PartialEq, uniffi::Enum)]
+pub enum DeviceMnemonicBuildResult {
```

Comment:
I would call this `DeviceMnemonicBuildOutcome`  - we try to always let `**Result` be a `Result` type.